### ISSUE TITLE
Docker compose

### ISSUE THREAD

**Issue by skovzhaw** [timescale/timescaledb#91]
Hey guys,

I am trying to bring up TimescaleDB via Docker Compose (as opposed to Docker run) and it seems to me that the extension is not being loaded.

**Dockerfile**
```Dockerfile
FROM timescale/timescaledb

ENV PGDATA=/var/lib/postgresql/data/timescaledb
VOLUME /var/lib/postgresql
EXPOSE 5432

ADD initialize.sh /docker-entrypoint-initdb.d/init-user-db.sh
```

**initialize.sh**
```Bash
#!/bin/bash

psql --username "$POSTGRES_USER" <<EOF
CREATE DATABASE my_db WITH OWNER $POSTGRES_USER;
GRANT ALL PRIVILEGES ON DATABASE my_db TO $POSTGRES_USER;

\c my_db
CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;

CREATE TABLE IF NOT EXISTS  my_table (
  time      TIMESTAMP         NOT NULL,
  metric    TEXT              NOT NULL,
  usage     DOUBLE PRECISION  NOT NULL
);
SELECT create_hypertable('my_table', 'time');
CREATE INDEX IF NOT EXISTS usage_metric ON my_table (metric, time DESC);
EOF
```

Compose.yml
```Yaml
version: '3'

services:
  timescale:
    image: my_timescale
    command: postgres -c shared_preload_libraries=timescaledb
    ports:
      - 5432:5432
    environment:
      - POSTGRES_USER=my_user
      - POSTGRES_PASSWORD=my_password
    volumes:
      - ./timescale:/var/lib/postgresql
```

As you can see I am trying to load _timescaledb_ extension via `command: postgres -c shared_preload_libraries=timescaledb` but it looks like nothing is happening. This is what I get as an output:

```
NOTICE:  installing required extension "postgres_fdw"
ERROR:  The timescaledb library is not preloaded
HINT:  Please preload the timescaledb library via shared_preload_libraries.

  This can be done by editing the config file at: /var/lib/postgresql/data/timescaledb/postgresql.conf
  and adding 'timescaledb' to the list in the shared_preload_libraries config.

    # Modify postgresql.conf:
      shared_preload_libraries = 'timescaledb'

  Another way to do this, if not preloading other libraries, is with the command:
  echo "shared_preload_libraries = 'timescaledb'" >> /var/lib/postgresql/data/timescaledb/postgresql.conf 

  (Will require a database restart.)

  If you REALLY know what you are doing and would like to load the library without preloading, you can disable   this check with: 
    SET timescaledb.allow_install_without_preload = 'on';

STATEMENT:  CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
```

And consequently that's why _create_hypertable_ fails.

Starting TimescaleDB via Docker run works for me without any issues. I also tried to put aftermentioned `postgres -c shared_preload_libraries=timescaledb` command to Dockerfile, but it did not help.

Am I missing something? How are you doing your Docker compose?

Thanks in advance.

Best wishes,
Martin

**Comment by Romathonat** [comment]
Hello,
I had the same problem and struggled with this way too long.
The probleme is that you add an init file to /docker-entrypoint-initdb.d : if you look at the [Dockerfile of postgres](https://github.com/docker-library/postgres/blob/54053ad27ac099abff3d4964bf7460fb9c541d5d/9.6/Dockerfile), you will see that this docker-entrypoint launch your init files **before** using timescaledb as a dependency !

In order to fix that, I created another container, expecially for the init of the db, it looks like this : 

```bash
FROM debian:jessie
RUN apt-get update && apt-get install -y netcat

# in order to have psql
RUN apt-get install -y postgresql-client

ADD ./init_schema.sql /
ADD ./start.sh /start.sh

CMD ["/start.sh"]
```

and the start.sh looks like this : 

``` bash
#!/bin/sh

while ! nc -z postgres 5432;do sleep 3;done

# now that the database is up, we can log into it and launch the init
PGPASSWORD=my password psql -h name_in_docker_compose -U user database< init_schema.sql

echo 'init done'

```
The part with the while is here to be sure the database is up before filling it with data.

```bash
version: '3'
services:
  postgres:
    build:
      context: ./postgres
    container_name: postgres

  postgres_init:
    build:
      context: ./postgres/postgres_init
    container_name: postgres_init
    depends_on:
      - postgres
```

I also put the command "postgres -c shared_preload_libraries=timescaledb" in the dockerfile as CMD [...] instead of the docker-compose.

This workaround works but it is not really clean, I admit, but it works until we find something better.

**Comment by RobAtticus** [comment]
Hi, we just released a new version (0.0.12-beta) that also included a new build process for Docker images (using the timescale/timescaledb-docker repo). This should have streamlined the build image, so maybe give it a try and see if it works better with Docker Compose? If not we can do some more investigating.

**Comment by Romathonat** [comment]
I tested it today, it works perfectly, good job. Now you can add you init scripts to /docker-entrypoint-initdb.d in the timescaledb docker, postgres will use it correctly (without saying that it can't use plugin timescaledb or whatever)
